cmake/OpenCVDetectCUDA.cmake

```diff
@@ -342,6 +342,10 @@ if(CUDA_FOUND)
     set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} --use_fast_math)
   endif()
 
+  if(MSVC)
+    OCV_OPTION(CUDA_ENABLE_DELAYLOAD "Enable delayed loading of CUDA DLLs" OFF)
+  endif()
```

Comment:
`OCV_OPTION` has `VISIBLE_IF` argument for conditions. I propose to replace the "if" with option with single line:
```
OCV_OPTION(CUDA_ENABLE_DELAYLOAD "Enable delayed loading of CUDA DLLs" OFF VISIBLE_IF MSVC AND (CMAKE_GENERATOR MATCHES "Visual Studio"))
```

Reply:
I prefer to have it optional. It may affect libraries loading order for guys why use other CUDA-based libraries like TensorRT, CuDNN, python code with CUDA.

Reply:
Alright, just committed your proposal. So the PR be merged now?